src/navigation/routes/AppStack.tsx

```diff
@@ -1,20 +1,46 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 import {createStackNavigator} from '@react-navigation/stack';
 import screenNames from '../ScreenNames';
 import AuthStack from './AuthStack';
+import {useDispatch, useSelector} from 'react-redux';
+import {AppRootStateType} from '../../types/types';
+import {setIsLoggedIn, setUser} from '../../store/actions/authAction';
+import auth from '@react-native-firebase/auth';
+import MainStack from './MainStack';
 
 const Stack = createStackNavigator<any>();
 
 export const AppStack = () => {
+  const dispatch = useDispatch();
+  const isLoggedIn = useSelector<AppRootStateType, boolean>(
```

Comment:
Create a separate file for selectors in store folder and import this selector from there. This will help you re-use duplicate selectors and make code more readable